ThirdPartyModules/Amasty/StoreCredit.php

```diff
@@ -111,4 +112,53 @@ public function filterProcessLayout($result)
         }
         return $result;
     }
+    
+    /**
+     * Return code if the quote has Amasty store credits.
+     * 
+     * @param $result
+     * @param $couponCode
+     * @param $quote
+     * 
+     * @return array
+     */
+    public function filterVerifyAppliedStoreCredit (
+        $result,
+        $couponCode,
+        $quote
+    )
+    {
+        if ($couponCode == self::AMASTY_STORECREDIT && $quote->getData(\Amasty\StoreCredit\Api\Data\SalesFieldInterface::AMSC_USE)) {
+            $result[] = $couponCode;
+        }
+        
+        return $result;
+    }
+    
+    /**
+     * Remove Amasty store credits from the quote.
+     *
+     * @param $amastyApplyStoreCreditToQuote
+     * @param $couponCode
+     * @param $quote
+     * @param $websiteId
+     * @param $storeId
+     * 
+     */
+    public function removeAppliedStoreCredit (
+        $amastyApplyStoreCreditToQuote,
+        $couponCode,
+        $quote,
+        $websiteId,
+        $storeId
+    )
+    {
+        try {
+            if ($couponCode == self::AMASTY_STORECREDIT && $quote->getData(\Amasty\StoreCredit\Api\Data\SalesFieldInterface::AMSC_USE)) {
+                $amastyApplyStoreCreditToQuote->cancel($quote->getId());
+            }
+        } catch (\Exception $e) {
+            throw $e;
```

Comment:
Can you log the exception to Bugsang?

Reply:
@johnmccombs1 It is there https://github.com/BoltApp/bolt-magento2/blob/master/Model/Api/UpdateCart.php#L298